### ISSUE TITLE
calico/node:v1.1.0-rc2 cyclically crashing on startup

### ISSUE THREAD

**Issue by lwr20** [projectcalico/calicoctl#1498]
System was setup according to http://docs.projectcalico.org/v2.0/getting-started/kubernetes/installation/hosted/k8s-backend/ on GCE, using `kubeadm init --pod-network-cidr=10.244.0.0/16`, then joining nodes and finally `kubectl apply -f calico.yaml` where the calico.yaml is the one from http://docs.projectcalico.org/v2.0/getting-started/kubernetes/installation/hosted/k8s-backend/calico.yaml but with the calico/node version changed to v1.1.0-rc2

Logs from calico-node pod:
```
$ kubectl logs calico-node-shjzn -n kube-system calico-node
time="2017-02-06T13:46:19Z" level=info msg="NODENAME environment not specified - check HOSTNAME" 
time="2017-02-06T13:46:19Z" level=info msg="Loading config from environment" 
Skipping datastore connection test
Calico is using a Kubernetes datastore
time="2017-02-06T13:46:19Z" level=info msg="Ensuring ThirdPartyResources exist" 
time="2017-02-06T13:46:19Z" level=info msg="Ensuring GlobalConfig ThirdPartyResource exists" 
time="2017-02-06T13:46:19Z" level=info msg="Ensuring IP Pool ThirdPartyResource exists" 
time="2017-02-06T13:46:19Z" level=info msg="ThirdPartyResources exist" 
time="2017-02-06T13:46:19Z" level=info msg="Ensuring ClusterType is set" 
time="2017-02-06T13:46:19Z" level=info msg="ClusterType is set" 
time="2017-02-06T13:46:19Z" level=info msg="Using previously configured cluster GUID" 
time="2017-02-06T13:46:19Z" level=info msg="Kubernetes is initialized as a Calico datastore" 
time="2017-02-06T13:46:19Z" level=info msg="Loading config from environment" 
panic: interface conversion: interface is nil, not string

goroutine 1 [running]:
panic(0x13b6680, 0xc4201b0fc0)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/projectcalico/calicoctl/vendor/github.com/projectcalico/libcalico-go/lib/client.(*config).getValue(0xc420024148, 0x1d8e380, 0xc4203d2ee0, 0x1d8e380, 0xc4203d2ee0, 0x0)
	/go/src/github.com/projectcalico/calicoctl/vendor/github.com/projectcalico/libcalico-go/lib/client/config.go:398 +0x15f
github.com/projectcalico/calicoctl/vendor/github.com/projectcalico/libcalico-go/lib/client.(*config).GetNodeIPIPTunnelAddress(0xc420024148, 0xc420010279, 0x1a, 0x0, 0x0, 0x0)
	/go/src/github.com/projectcalico/calicoctl/vendor/github.com/projectcalico/libcalico-go/lib/client/config.go:191 +0xe7
main.removeHostTunnelAddr(0xc420353ae0, 0xc420010279, 0x1a)
	/go/src/github.com/projectcalico/calicoctl/calico_node/allocateipip/allocate_ipip_addr.go:78 +0x8e
main.main()
	/go/src/github.com/projectcalico/calicoctl/calico_node/allocateipip/allocate_ipip_addr.go:37 +0xed
Calico node failed to start
```

```
$ kubectl describe po -n kube-system calico-node-shjzn
Name:		calico-node-shjzn
Namespace:	kube-system
Node:		tigera-lwr-kubetest-master/10.240.0.27
Start Time:	Mon, 06 Feb 2017 13:40:23 +0000
Labels:		k8s-app=calico-node
Status:		Running
IP:		10.240.0.27
Controllers:	DaemonSet/calico-node
Containers:
  calico-node:
    Container ID:	docker://777eea0646814a11bf45eb2ef7e3e62c2720e6601dc9f94e41a8f2adc4e3ad20
    Image:		calico/node:v1.1.0-rc2
    Image ID:		docker-pullable://calico/node@sha256:390dd7934798dfddb46a62d08da2614de9c338cd732c0a4d1e7b3d7030e6bc63
    Port:		
    State:		Terminated
      Reason:		Error
      Exit Code:	1
      Started:		Mon, 06 Feb 2017 13:46:19 +0000
      Finished:		Mon, 06 Feb 2017 13:46:19 +0000
    Last State:		Terminated
      Reason:		Error
      Exit Code:	1
      Started:		Mon, 06 Feb 2017 13:43:33 +0000
      Finished:		Mon, 06 Feb 2017 13:43:33 +0000
    Ready:		False
    Restart Count:	6
    Volume Mounts:
      /lib/modules from lib-modules (ro)
      /var/run/calico from var-run-calico (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-98tcs (ro)
    Environment Variables:
      DATASTORE_TYPE:				kubernetes
      FELIX_LOGSEVERITYSCREEN:			debug
      CALICO_NETWORKING_BACKEND:		none
      CALICO_DISABLE_FILE_LOGGING:		true
      FELIX_DEFAULTENDPOINTTOHOSTACTION:	ACCEPT
      HOSTNAME:					 (v1:spec.nodeName)
      IP:					
  install-cni:
    Container ID:	docker://a3d0ca5655a569513efbaca074db595b841016171a3deb19e86d2709719c717b
    Image:		calico/cni:v1.5.6
    Image ID:		docker-pullable://calico/cni@sha256:d0e07c19aac1cc84278decd7e24e8cc1842bfd2332037c652c81f525d555052d
    Port:		
    Command:
      /install-cni.sh
    State:		Running
      Started:		Mon, 06 Feb 2017 13:40:34 +0000
    Ready:		True
    Restart Count:	0
    Volume Mounts:
      /host/etc/cni/net.d from cni-net-dir (rw)
      /host/opt/cni/bin from cni-bin-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-98tcs (ro)
    Environment Variables:
      CNI_NETWORK_CONFIG:	<set to the key 'cni_network_config' of config map 'calico-config'>
      KUBERNETES_NODE_NAME:	 (v1:spec.nodeName)
Conditions:
  Type		Status
  Initialized 	True 
  Ready 	False 
  PodScheduled 	True 
Volumes:
  lib-modules:
    Type:	HostPath (bare host directory volume)
    Path:	/lib/modules
  var-run-calico:
    Type:	HostPath (bare host directory volume)
    Path:	/var/run/calico
  cni-bin-dir:
    Type:	HostPath (bare host directory volume)
    Path:	/opt/cni/bin
  cni-net-dir:
    Type:	HostPath (bare host directory volume)
    Path:	/etc/cni/net.d
  default-token-98tcs:
    Type:	Secret (a volume populated by a Secret)
    SecretName:	default-token-98tcs
QoS Class:	BestEffort
Tolerations:	CriticalAddonsOnly=:Exists
		dedicated=master:NoSchedule
Events:
  FirstSeen	LastSeen	Count	From					SubobjectPath			Type		Reason		Message
  ---------	--------	-----	----					-------------			--------	------		-------
  5m		5m		1	{kubelet tigera-lwr-kubetest-master}	spec.containers{calico-node}	Normal		Pulling		pulling image "calico/node:v1.1.0-rc2"
  5m		5m		1	{kubelet tigera-lwr-kubetest-master}	spec.containers{calico-node}	Normal		Pulled		Successfully pulled image "calico/node:v1.1.0-rc2"
  5m		5m		1	{kubelet tigera-lwr-kubetest-master}	spec.containers{calico-node}	Normal		Started		Started container with docker id 1d53f536d5ce
  5m		5m		1	{kubelet tigera-lwr-kubetest-master}	spec.containers{install-cni}	Normal		Pulling		pulling image "calico/cni:v1.5.6"
  5m		5m		1	{kubelet tigera-lwr-kubetest-master}	spec.containers{calico-node}	Normal		Created		Created container with docker id 1d53f536d5ce; Security:[seccomp=unconfined]
  5m		5m		1	{kubelet tigera-lwr-kubetest-master}	spec.containers{install-cni}	Normal		Created		Created container with docker id a3d0ca5655a5; Security:[seccomp=unconfined]
  5m		5m		1	{kubelet tigera-lwr-kubetest-master}	spec.containers{install-cni}	Normal		Pulled		Successfully pulled image "calico/cni:v1.5.6"
  5m		5m		1	{kubelet tigera-lwr-kubetest-master}	spec.containers{install-cni}	Normal		Started		Started container with docker id a3d0ca5655a5
  5m		5m		1	{kubelet tigera-lwr-kubetest-master}	spec.containers{calico-node}	Normal		Created		Created container with docker id 79f987e7e37e; Security:[seccomp=unconfined]
  5m		5m		1	{kubelet tigera-lwr-kubetest-master}	spec.containers{calico-node}	Normal		Started		Started container with docker id 79f987e7e37e
  5m		5m		1	{kubelet tigera-lwr-kubetest-master}					Warning		FailedSync	Error syncing pod, skipping: failed to "StartContainer" for "calico-node" with CrashLoopBackOff: "Back-off 10s restarting failed container=calico-node pod=calico-node-shjzn_kube-system(cf6462b7-ec71-11e6-ba48-42010af0001b)"

  5m	5m	1	{kubelet tigera-lwr-kubetest-master}	spec.containers{calico-node}	Normal	Created		Created container with docker id 029d8e5f4739; Security:[seccomp=unconfined]
  5m	5m	1	{kubelet tigera-lwr-kubetest-master}	spec.containers{calico-node}	Normal	Started		Started container with docker id 029d8e5f4739
  5m	5m	2	{kubelet tigera-lwr-kubetest-master}					Warning	FailedSync	Error syncing pod, skipping: failed to "StartContainer" for "calico-node" with CrashLoopBackOff: "Back-off 20s restarting failed container=calico-node pod=calico-node-shjzn_kube-system(cf6462b7-ec71-11e6-ba48-42010af0001b)"
```

**Comment by fasaxc** [comment]
New k8s backend returning nil rather than the expected "key not found" error?

**Comment by caseydavenport** [comment]
Yes, looks like the error is here https://github.com/projectcalico/libcalico-go/blob/master/lib/backend/k8s/k8s.go#L720

Should be returning `errors.ErrorResourceDoesNotExist` instead.

**Comment by heschlie** [comment]
fixed in projectcalico/libcalico-go#339
